backend/src/test/java/de/neuefische/backend/controller/UserControllerTest.java

```diff
@@ -89,23 +89,64 @@ void testPostNewUser_shouldReturn_409_when_username_exists() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/api/user/register")
+                        .contentType("application/json")
+                        .content("""
+                                {
+                                        "username": "user@event.hub",
+                                        "password": "123"
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(status().is(201))
+                .andExpect(content().json("""
+                        {
+                                        "username": "user@event.hub",
+                                        "roles": [
+                                                "user"
+                                        ]
+                        }
+                        """))
+                .andExpect(jsonPath("$.id").isNotEmpty());
+
+        mockMvc.perform((MockMvcRequestBuilders.put("/api/user/update-preferred-categories")
+                        .contentType("application/json")
+                        .content("""
+                                {
+                                        "username": "user@event.hub",
+                                        "categories": ["MUSIC", "SPORTS"]
+                                }
+                                """)
+                        .with(csrf())))
+                .andExpect(status().is(200))
+                .andExpect(content().json("""
+                        {
+                                        "username": "user@event.hub",
+                                        "roles": [
+                                                "user"
+                                        ],
+                                        "preferredCategories": ["MUSIC", "SPORTS"]
+                        }
+                        """));
+
         mockMvc.perform(MockMvcRequestBuilders.post("/api/user/login")
                         .contentType("application/json")
                         .with(csrf()))
                 .andExpect(status().is(200))
                 .andExpect(content().json("""
                         {
                                         "username": "user@event.hub",
-                                        "roles": ["ROLE_USER"]
+                                        "roles": ["ROLE_USER"],
+                                        "preferredCategories": ["MUSIC", "SPORTS"]
                         }
                         """));
     }
 
     @Test
     @DirtiesContext
     @WithMockUser
-    void logout_should_invalidate_session_and_clear_context() throws Exception {
+    void testLogout_should_invalidate_session_and_clear_context() throws Exception {
         mockMvc.perform(MockMvcRequestBuilders.post("/api/user/logout")
                         .with(csrf()))
                 .andExpect(MockMvcResultMatchers.status().is(200));
```

Comment:
No need to have MockMvcResultMatchers. here. You can just delete it because it is imported. Just like the other tests.